x-pack/plugins/lens/public/visualizations/xy/types.ts

```diff
@@ -277,9 +278,9 @@ export const visualizationTypes: VisualizationType[] = [
 
 interface XYStateWithLayers {
   [prop: string]: unknown;
-  layers: XYLayerConfig[];
+  layers: Record<string, FormBasedLayer>;
```

Comment:
This doesn't look right - `FormBasedLayer` is _very_ different than `XYLayerConfig`

Reply:
It was wrong before. It's interface about `datasourceState`. Maybe we should rename it.